chain/client/src/sync/epoch.rs

```diff
@@ -111,17 +111,51 @@ impl EpochSync {
             .get_ser::<EpochInfo>(DBCol::EpochInfo, next_epoch.0.as_bytes())?
             .ok_or_else(|| Error::EpochOutOfBounds(next_epoch))?;
 
-        // TODO: don't always generate from genesis
-        let all_past_epochs = Self::get_all_past_epochs(
+        let genesis_epoch_info = store
+            .get_ser::<EpochInfo>(DBCol::EpochInfo, EpochId::default().0.as_bytes())?
+            .ok_or_else(|| Error::EpochOutOfBounds(EpochId::default()))?;
+
+        // If we have an existing (possibly and likely outdated) EpochSyncProof stored on disk,
+        // the last epoch we have a proof for is the "previous epoch" included in that EpochSyncProof.
+        // Otherwise, the last epoch we have a "proof" for is the genesis epoch.
+        let existing_epoch_sync_proof =
```

Comment:
I can see that the receiving node is storing the epoch sync proof but shouldn't the current node store the updated proof as well after generating it in `derive_epoch_sync_proof`?

Reply:
Good question. I argue that it should *not*. The reason is that if we need to fix this or change this for some reason in the future, e.g. needing some new field because of a protocol upgrade, we would not have to do a database migration.

The nodes that bootstrapped via epoch sync have no *choice* but to store it, because the EpochSyncProof represents the gap in history that the new node does not have.